File: Facultate/Anul 2/Sem 2/MPP/Laburi/Lab_2_All/CSharpLab_2/CSharpLab_2/models/Ride.cs
namespace CSharpLab_2.models;

public class Ride : Entity<long>
{
    public required string Destination { get; init; }
    public required DateOnly Date { get; init; }
    public required TimeOnly DepartureTime { get; init; }
    public int AvailableSeats { get; set; } = 18;
}

File: Facultate/Anul 2/Sem 2/MPP/Laburi/Lab3All/JavaLab3/src/main/java/models/Ride.java
package models;

import java.time.LocalDate;
import java.time.LocalTime;

public class Ride extends Entity<Long>{

    private final String destination;
    private final LocalDate date;
    private final LocalTime departureTime;
    private Integer availableSeats = 18;

    public Ride(String destination, LocalDate date, LocalTime departureTime) {
        this.destination = destination;
        this.date = date;
        this.departureTime = departureTime;
    }

    public String getDestination() {
        return destination;
    }

    public LocalDate getDate() {
        return date;
    }

    public LocalTime getDepartureTime() {
        return departureTime;
    }

    public Integer getAvailableSeats() {
        return availableSeats;
    }

    public void setAvailableSeats(Integer availableSeats) {
        this.availableSeats = availableSeats;
    }
}
